### ISSUE TITLE
$Config{incpath} and $Config{libpth} are garbage paths on >= VC 2005

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#14409]
Migrated from [rt.perl.org#123575](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=123575) (status was 'open')

Searchable as RT123575$

**Comment by p5pRT** [comment]
### From @bulk88

#### Created by @bulk88




VC 2005 and newer don't set the MSVCDIR env var\. There is a comment from 
1998 
http&#8203;://perl5\.git\.perl\.org/perl\.git/commitdiff/95883bf30d6fc6c143368c6f07aea00e33c76a2b 
about CCHOME being broken sometimes\, but my VC 6 and my VC 2003 both set 
MSVCDIR \(neither set CCHOME which I think is a unix/perl define\, not a 
MS define\)\. My 2005 and 2008 do not\. VCINSTALLDIR is the replacement for 
MSVCDIR env var for >= 2005\.

chunk of win32/makefile that is the problem \(makefile\.mk will also need 
fixing\)

\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-

\#
\#
\# set the install locations of the compiler include/libraries
\# Running VCVARS32\.BAT is \*required\* when using Visual C\.
\# Some versions of Visual C don't define MSVCDIR in the environment\,
\# so you may have to set CCHOME explicitly \(spaces in the path name should
\# not be quoted\)
\#
CCHOME        = $\(MSVCDIR\)
CCINCDIR    = $\(CCHOME\)\\include
CCLIBDIR    = $\(CCHOME\)\\lib

\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-




ICL with VC 2008

\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
C&#8203;:\\p521\\srcnew>perl \-MConfig \-E" say $Config&#8203;::Config\{incpath\}
\\include

C&#8203;:\\p521\\srcnew>perl \-MConfig \-E" say $Config&#8203;::Config\{libpth\}
\\lib

C&#8203;:\\p521\\srcnew>perl \-V
Summary of my perl5 \(revision 5 version 21 subversion 8\) configuration&#8203;:
&nbsp;   Derived from&#8203;:
&nbsp;   Platform&#8203;:
&nbsp;     osname=MSWin32\, osvers=5\.2\, archname=MSWin32\-x86\-multi\-thread
&nbsp;     uname=''
&nbsp;     config\_args='undef'
&nbsp;     hint=recommended\, useposix=true\, d\_sigaction=undef
&nbsp;     useithreads=define\, usemultiplicity=define
&nbsp;     use64bitint=undef\, use64bitall=undef\, uselongdouble=undef
&nbsp;     usemymalloc=n\, bincompat5005=undef
&nbsp;   Compiler&#8203;:
&nbsp;     cc='icl'\, ccflags ='\-nologo \-GF \-W3 \-Qstd=c99 \-O1 \-MD \-Zi \-DNDEBUG 
\-GS\- \-GL
\-DWIN32 \-D\_CONSOLE \-DNO\_STRICT \-D\_CRT\_SECURE\_NO\_DEPRECATE 
\-D\_CRT\_NONSTDC\_NO\_DEPR
ECATE  \-DPERL\_TEXTMODE\_SCRIPTS \-DPERL\_HASH\_FUNC\_ONE\_AT\_A\_TIME 
\-DNO\_MATHOMS \-DPER
L\_IMPLICIT\_CONTEXT \-DPERL\_IMPLICIT\_SYS \-DUSE\_PERLIO'\,
&nbsp;     optimize='\-O1 \-MD \-Zi \-DNDEBUG \-GS\- \-GL'\,
&nbsp;     cppflags='\-DWIN32'
&nbsp;     ccversion='12\.1\.5\.344'\, gccversion=''\, gccosandvers=''
&nbsp;     intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234\, 
doublekind=3

&nbsp;     d\_longlong=undef\, longlongsize=8\, d\_longdbl=define\, longdblsize=8\, 
longdblki
nd=0
&nbsp;     ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, 
Off\_t='\_\_int64'\, lseeksi
ze=8
&nbsp;     alignbytes=8\, prototype=define
&nbsp;   Linker and Libraries&#8203;:
&nbsp;     ld='xilink'\, ldflags ='\-nologo \-nodefaultlib \-debug \-opt&#8203;:ref\,icf 
\-ltcg  \-lib
path&#8203;:"c&#8203;:\\p521\\lib\\CORE"  \-machine&#8203;:x86 "/manifestdependency&#8203;:type='Win32' 
name='Mi
crosoft\.Windows\.Common\-Controls' version='6\.0\.0\.0' 
processorArchitecture='\*' pub
licKeyToken='6595b64144ccf1df' language='\*'"'
&nbsp;     libpth=\\lib
&nbsp;     libs=oldnames\.lib kernel32\.lib user32\.lib gdi32\.lib winspool\.lib 
comdlg32\.l
ib advapi32\.lib shell32\.lib ole32\.lib oleaut32\.lib  netapi32\.lib 
uuid\.lib ws2\_32
\.lib mpr\.lib winmm\.lib  version\.lib odbc32\.lib odbccp32\.lib comctl32\.lib 
libircm
t\.lib libmmd\.lib msvcrt\.lib
&nbsp;     perllibs=oldnames\.lib kernel32\.lib user32\.lib gdi32\.lib 
winspool\.lib  comdlg
32\.lib advapi32\.lib shell32\.lib ole32\.lib oleaut32\.lib  netapi32\.lib 
uuid\.lib ws
2\_32\.lib mpr\.lib winmm\.lib  version\.lib odbc32\.lib odbccp32\.lib 
comctl32\.lib lib
ircmt\.lib libmmd\.lib msvcrt\.lib
&nbsp;     libc=msvcrt\.lib\, so=dll\, useshrplib=true\, libperl=perl521\.lib
&nbsp;     gnulibc\_version=''
&nbsp;   Dynamic Linking&#8203;:
&nbsp;     dlsrc=dl\_win32\.xs\, dlext=dll\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;     cccdlflags=' '\, lddlflags='\-dll \-nologo \-nodefaultlib \-debug 
\-opt&#8203;:ref\,icf \-l
tcg  \-libpath&#8203;:"c&#8203;:\\p521\\lib\\CORE"  \-machine&#8203;:x86 
"/manifestdependency&#8203;:type='Win32'
&nbsp;  name='Microsoft\.Windows\.Common\-Controls' version='6\.0\.0\.0' 
processorArchitectur
e='\*' publicKeyToken='6595b64144ccf1df' language='\*'"'


Characteristics of this binary \(from libperl\)&#8203;:
&nbsp;   Compile\-time options&#8203;: HAS\_TIMES HAVE\_INTERP\_INTERN MULTIPLICITY 
NO\_MATHOMS
&nbsp;                         PERLIO\_LAYERS PERL\_DISABLE\_PMC 
PERL\_DONT\_CREATE\_GVSV
&nbsp;                         PERL\_HASH\_FUNC\_ONE\_AT\_A\_TIME PERL\_IMPLICIT\_CONTEXT
&nbsp;                         PERL\_IMPLICIT\_SYS PERL\_NEW\_COPY\_ON\_WRITE
&nbsp;                         PERL\_PRESERVE\_IVUV USE\_ITHREADS USE\_LARGE\_FILES
&nbsp;                         USE\_LOCALE USE\_LOCALE\_COLLATE USE\_LOCALE\_CTYPE
&nbsp;                         USE\_LOCALE\_NUMERIC USE\_LOCALE\_TIME USE\_PERLIO
&nbsp;                         USE\_PERL\_ATOF
&nbsp;   Locally applied patches&#8203;:
&nbsp;         uncommitted\-changes
&nbsp;   Built under MSWin32
&nbsp;   Compiled at Jan  7 2015 16&#8203;:03&#8203;:45
&nbsp;   @&#8203;INC&#8203;:
&nbsp;     C&#8203;:/p521/srcnew/lib
&nbsp;     \.

C&#8203;:\\p521\\srcnew>
\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
C&#8203;:\\>perl \-MConfig \-E" say $Config&#8203;::Config\{incpath\}
"C&#8203;:\\Program Files\\Microsoft Visual Studio \.NET 2003\\VC7\\include"

C&#8203;:\\>perl \-MConfig \-E" say $Config&#8203;::Config\{libpth\}
"C&#8203;:\\Program Files\\Microsoft Visual Studio \.NET 2003\\VC7\\lib"

C&#8203;:\\>perl \-V
Summary of my perl5 \(revision 5 version 21 subversion 8\) configuration&#8203;:
&nbsp;   Derived from&#8203;: 931bdd676fc5e5393569d4db0f71508204973353
&nbsp;   Ancestor&#8203;: 735ecbe919d740c2d0068de6daf6f2359bbcd3eb
&nbsp;   Platform&#8203;:
&nbsp;     osname=MSWin32\, osvers=5\.1\, archname=MSWin32\-x86\-multi\-thread
&nbsp;     uname=''
&nbsp;     config\_args='undef'
&nbsp;     hint=recommended\, useposix=true\, d\_sigaction=undef
&nbsp;     useithreads=define\, usemultiplicity=define
&nbsp;     use64bitint=undef\, use64bitall=undef\, uselongdouble=undef
&nbsp;     usemymalloc=n\, bincompat5005=undef
&nbsp;   Compiler&#8203;:
&nbsp;     cc='cl'\, ccflags ='\-nologo \-GF \-W3 \-O1 \-MD \-Zi \-DNDEBUG \-G7 \-GL 
\-DWIN32 \-D\_C
ONSOLE \-DNO\_STRICT  \-DPERL\_TEXTMODE\_SCRIPTS 
\-DPERL\_HASH\_FUNC\_ONE\_AT\_A\_TIME \-DNO\_
MATHOMS \-DPERL\_IMPLICIT\_CONTEXT \-DPERL\_IMPLICIT\_SYS \-DUSE\_PERLIO 
\-D\_USE\_32BIT\_TI
ME\_T'\,
&nbsp;     optimize='\-O1 \-MD \-Zi \-DNDEBUG \-G7 \-GL'\,
&nbsp;     cppflags='\-DWIN32'
&nbsp;     ccversion='13\.10\.6030'\, gccversion=''\, gccosandvers=''
&nbsp;     intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234\, 
doublekind=3

&nbsp;     d\_longlong=undef\, longlongsize=8\, d\_longdbl=define\, longdblsize=8\, 
longdblki
nd=0
&nbsp;     ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, 
Off\_t='\_\_int64'\, lseeksi
ze=8
&nbsp;     alignbytes=8\, prototype=define
&nbsp;   Linker and Libraries&#8203;:
&nbsp;     ld='link'\, ldflags ='\-nologo \-nodefaultlib \-debug \-opt&#8203;:ref\,icf 
\-ltcg  \-libpa
th&#8203;:"c&#8203;:\\perl521\\lib\\CORE"  \-machine&#8203;:x86'
&nbsp;     libpth="C&#8203;:\\Program Files\\Microsoft Visual Studio \.NET 2003\\VC7\\lib"
&nbsp;     libs=oldnames\.lib kernel32\.lib user32\.lib gdi32\.lib winspool\.lib 
comdlg32\.l
ib advapi32\.lib shell32\.lib ole32\.lib oleaut32\.lib  netapi32\.lib 
uuid\.lib ws2\_32
\.lib mpr\.lib winmm\.lib  version\.lib odbc32\.lib odbccp32\.lib comctl32\.lib 
msvcrt\.
lib
&nbsp;     perllibs=oldnames\.lib kernel32\.lib user32\.lib gdi32\.lib 
winspool\.lib  comdlg
32\.lib advapi32\.lib shell32\.lib ole32\.lib oleaut32\.lib  netapi32\.lib 
uuid\.lib ws
2\_32\.lib mpr\.lib winmm\.lib  version\.lib odbc32\.lib odbccp32\.lib 
comctl32\.lib msv
crt\.lib
&nbsp;     libc=msvcrt\.lib\, so=dll\, useshrplib=true\, libperl=perl521\.lib
&nbsp;     gnulibc\_version=''
&nbsp;   Dynamic Linking&#8203;:
&nbsp;     dlsrc=dl\_win32\.xs\, dlext=dll\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;     cccdlflags=' '\, lddlflags='\-dll \-nologo \-nodefaultlib \-debug 
\-opt&#8203;:ref\,icf \-l
tcg  \-libpath&#8203;:"c&#8203;:\\perl521\\lib\\CORE"  \-machine&#8203;:x86'


Characteristics of this binary \(from libperl\)&#8203;:
&nbsp;   Compile\-time options&#8203;: HAS\_TIMES HAVE\_INTERP\_INTERN MULTIPLICITY 
NO\_MATHOMS
&nbsp;                         PERLIO\_LAYERS PERL\_DISABLE\_PMC 
PERL\_DONT\_CREATE\_GVSV
&nbsp;                         PERL\_HASH\_FUNC\_ONE\_AT\_A\_TIME PERL\_IMPLICIT\_CONTEXT
&nbsp;                         PERL\_IMPLICIT\_SYS PERL\_NEW\_COPY\_ON\_WRITE
&nbsp;                         PERL\_PRESERVE\_IVUV USE\_ITHREADS USE\_LARGE\_FILES
&nbsp;                         USE\_LOCALE USE\_LOCALE\_COLLATE USE\_LOCALE\_CTYPE
&nbsp;                         USE\_LOCALE\_NUMERIC USE\_LOCALE\_TIME USE\_PERLIO
&nbsp;                         USE\_PERL\_ATOF
&nbsp;   Locally applied patches&#8203;:
&nbsp;         uncommitted\-changes
&nbsp;         930bcafb4bedf1eee527897a79ad7b62c51f2b4f
&nbsp;         474cae1469011a6e74d3802325bb199424f5d7c9
&nbsp;         2d246912ee0f281fd09b36b474421360a15dbb19
&nbsp;         f79f05b941f486f6760ae8ec13f775a09ecfcaba
&nbsp;         7eaa8b820ba9c96d567484d7daef50e6dc4f02e2
&nbsp;         fb36b548ca4a754d0f76aa947a83f9f6113f1f84
&nbsp;         ac71298f6ca37cb1aeaa56a5a575ba02b21b0146
&nbsp;         7de5101be126402e74b2c4365bad4f80963db881
&nbsp;         905ed8d12c70599234c855d668a9889ce14125c1
&nbsp;         971112bc4f2054d3d54c3888939d597c6457d475
&nbsp;         6a47f6bd25f2f61ccbdd81ec707e79864d456f38
&nbsp;         40499d7a4c53a9fa26cf348c39c9b460ccc53ae5
&nbsp;         093a069a3443d289d8fc0b2891b7f4197fb12759
&nbsp;         119fe3c390384aedd1f2030fea816e281721a7d8
&nbsp;         84399f7fd173d39c78fd474b9c6f1c2c4c0bd574
&nbsp;         bfb2ad625fc0ee2466c595bc354d907e087b1ac3
&nbsp;         36d0f17cb6067c880b214b9552688b0debf97db6
&nbsp;         107ca57883b3ce5b63722d8539f02226a340af0f
&nbsp;         9e8b0be5ca142765901afe60003b9298073f62c5
&nbsp;         931bdd676fc5e5393569d4db0f71508204973353
&nbsp;   Built under MSWin32
&nbsp;   Compiled at Jan  8 2015 07&#8203;:46&#8203;:26
&nbsp;   %ENV&#8203;:
&nbsp;     PERL\_JSON\_BACKEND="Cpanel&#8203;::JSON&#8203;::XS"
&nbsp;     PERL\_YAML\_BACKEND="YAML&#8203;::XS"
&nbsp;   @&#8203;INC&#8203;:
&nbsp;     C&#8203;:/perl521/site/lib
&nbsp;     C&#8203;:/perl521/lib
&nbsp;     \.

C&#8203;:\\>
\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-

Even though README\.w32 says CCHOME has to be set by hand I've never ever 
done it\, and I dont expect to start doing it now \(switching CCTYPE is 
enough work\)

\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
You will have to make sure that CCTYPE is set correctly and that
CCHOME points to wherever you installed your compiler\.
\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
The default value for CCHOME in the makefiles for Visual C\+\+
may not be correct for some versions\.  Make sure the default exists
and is valid\.
\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
\(The PlatformSDKDir might need to be set differently depending on which 
version
you are using\. Earlier versions installed into "C&#8203;:\\Program 
Files\\Microsoft SDK"\,
while the latest versions install into version\-specific locations such as
"C&#8203;:\\Program Files\\Microsoft Platform SDK for Windows Server 2003 R2"\.\)

Perl should now build using the win32/Makefile\.  You will need to edit that
file to set

&nbsp;    CCTYPE = MSVC80FREE

and to set CCHOME\, CCINCDIR and CCLIBDIR as per the environment setup above\.
\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-

<details><summary>Perl Info</summary>

```

Flags:
                      category=core
                      severity=low

Site configuration information for perl 5.21.7:

Configured by Owner at Sat Nov 22 21:54:54 2014.

Summary of my perl5 (revision 5 version 21 subversion 7) configuration:
                    Local Commit: 1bce52df028aabe28c20b2d97949e35c17ea811e
                    Ancestor: 7072da8afeba4c87ae623cd913e274396ffcf1cd
                    Platform:
                      osname=MSWin32, osvers=5.1,
archname=MSWin32-x86-multi-thread
                      uname=''
                      config_args='undef'
                      hint=recommended, useposix=true, d_sigaction=undef
                      useithreads=define, usemultiplicity=define
                      use64bitint=undef, use64bitall=undef,
uselongdouble=undef
                      usemymalloc=n, bincompat5005=undef
                    Compiler:
                      cc='cl', ccflags ='-nologo -GF -W3 -O1 -MD -Zi 
-DNDEBUG
-G7 -GL
-DWIN32 -D_CONSOLE -DNO_STRICT  -DPERL_TEXTMODE_SCRIPTS
-DPERL_HASH_FUNC_ONE_AT_A_TIME -DNO_MATHOMS -DPERL_IMPLICIT_CONTEXT
-DPERL_IMPLICIT_SYS -DUSE_PERLIO -D_USE_32BIT_TIME_T',
                      optimize='-O1 -MD -Zi -DNDEBUG -G7 -GL',
                      cppflags='-DWIN32'
                      ccversion='13.10.6030', gccversion='', gccosandvers=''
                      intsize=4, longsize=4, ptrsize=4, doublesize=8,
byteorder=1234,
doublekind=3
                      d_longlong=undef, longlongsize=8, d_longdbl=define,
longdblsize=8,
longdblkind=0
                      ivtype='long', ivsize=4, nvtype='double', nvsize=8,
Off_t='__int64',
lseeksize=8
                      alignbytes=8, prototype=define
                    Linker and Libraries:
                      ld='link', ldflags ='-nologo -nodefaultlib -debug
-opt:ref,icf
-ltcg  -libpath:"c:\perl521\lib\CORE"  -machine:x86'
                      libpth="C:\Program Files\Microsoft Visual Studio .NET
2003\VC7\lib"
                      libs=oldnames.lib kernel32.lib user32.lib gdi32.lib
winspool.lib
comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib
netapi32.lib uuid.lib ws2_32.lib mpr.lib winmm.lib  version.lib
odbc32.lib odbccp32.lib comctl32.lib msvcrt.lib
                      perllibs=oldnames.lib kernel32.lib user32.lib 
gdi32.lib
winspool.lib  comdlg32.lib advapi32.lib shell32.lib ole32.lib
oleaut32.lib  netapi32.lib uuid.lib ws2_32.lib mpr.lib winmm.lib
version.lib odbc32.lib odbccp32.lib comctl32.lib msvcrt.lib
                      libc=msvcrt.lib, so=dll, useshrplib=true,
libperl=perl521.lib
                      gnulibc_version=''
                    Dynamic Linking:
                      dlsrc=dl_win32.xs, dlext=dll, d_dlsymun=undef,
ccdlflags=' '
                      cccdlflags=' ', lddlflags='-dll -nologo -nodefaultlib
-debug
-opt:ref,icf -ltcg  -libpath:"c:\perl521\lib\CORE"  -machine:x86'

Locally applied patches:
                      ce7a4d57d0acca9f39a84d36d708c4505dfe45ca
                      ca0b263f4b167ddf97416f657d79ab5bd3344357
                      08919bf863666074243240abbd19cd1a74cc7b74
                      b8a043377dbf39548709b107a11e5cc2714c0e9a
                      efa855eb5cffb7739616c295dd968d1510efeeb0
                      1d47d0b810e26d9a2f9101fb813bd5b3dd332cc9
                      3faca062ddb056db54f73fa55b0a9d473675dd33
                      0b3e905bda3e75ad948a1213f620656b60807393
                      1b1efc719fde05d215e5a13fb38c03e12a3aab08
                      1bce52df028aabe28c20b2d97949e35c17ea811e


@INC for perl 5.21.7:
                      ..\lib
                      C:/perl521/srcnewb4opt/lib
                      .


Environment for perl 5.21.7:
                      HOME (unset)
                      LANG (unset)
                      LANGUAGE (unset)
                      LD_LIBRARY_PATH (unset)
                      LOGDIR (unset)
                      PATH=C:\WINDOWS\system32;C:\Program Files\Microsoft
Visual
Studio
.NET 2003\Vc7\bin;C:\Program Files\Microsoft Visual Studio .NET
2003\Common7\IDE;C:\WINDOWS;C:\Program Files\Git\cmd;C:\Program
Files\Microsoft Visual Studio .NET 2003\Common7\Tools\bin;C:\perl\bin
                      PERL_BADLANG (unset)
                      PERL_JSON_BACKEND=Cpanel::JSON::XS
                      PERL_YAML_BACKEND=YAML
                      SHELL (unset)
























```
</details>

**Comment by p5pRT** [comment]
### From @bulk88

Since there have been no comments for this bug\, I have written a patch for review\. Mingw is likely to fail
the added tests so that is why I added WIP\.

\-\- 
bulk88 ~ bulk88 at hotmail\.com

**Comment by p5pRT** [comment]
### From @bulk88

<details><summary>0001-WIP-perl-123575-make-sure-Config-incpath-and-Config-.patch</summary>

```diff
From 73620a5de4d991017a0256b4f52a2a1a4f6ca185 Mon Sep 17 00:00:00 2001
From: Daniel Dragan <bulk88@hotmail.com>
Date: Tue, 28 Apr 2015 20:26:13 -0400
Subject: [PATCH] WIP [perl #123575] make sure $Config{incpath} and
 $Config{libpth} are valid

VC 2005 and up used a different env var, and these 2 were being set to
"/include" and "/lib", which aren't valid paths on Win32. Fix it so newer
VC's are correctly looked up. libpth is mainly used by
ExtUtils::Liblist::Kid and it usually warns on VC Perl that it couldn't
absolute path the libraries when EUMM building and specifying additional
libs to EUMM to link with. Despite the warnings, linking still succeeds
since VC typically uses env vars to find standard c INCLUDE and LIB,
not cmd line args). Unix Perl typically leaves libpth and incpath as empty
string, so skip the tests if it is empty string.

Mingw is likely to fail the tests unless CCHOME is changed manually, which
it didn't need to be changed previously.
---
 lib/Config.t      | 24 ++++++++++++++++++++++++
 win32/Makefile    |  6 ++++++
 win32/makefile.mk |  8 +++++++-
 3 files changed, 37 insertions(+), 1 deletion(-)

diff --git a/lib/Config.t b/lib/Config.t
index abd9316..751abc0 100644
--- a/lib/Config.t
+++ b/lib/Config.t
@@ -277,3 +277,27 @@ foreach my $lib (qw(applibexp archlibexp privlibexp sitearchexp sitelibexp
   }
 }
 _diag ('@INC is:', @orig_inc) if $failed;
+
+#can't use Text::ParseWords in this .t
+sub extract_path {
+    my $paths = shift;
+    if(substr($paths,0,1) eq '"') {
+	return substr($paths, 1, index($paths, '"', 1));
+    }
+    else {
+	my $path_end = index($paths, ' ');
+	$path_end = length $paths if $path_end == -1;
+	return substr($paths, 0, $path_end);
+    }
+}
+#empty string on some platforms
+SKIP: {
+    skip '$Config{incpath} is empty string' if $Config{incpath} eq '';
+    my $path = extract_path($Config{incpath});
+    ok(-d $path, '$Config{incpath} is a valid dir');
+}
+SKIP: {
+    skip '$Config{libpth} is empty string' if $Config{libpth} eq '';
+    my $path = extract_path($Config{libpth});
+    ok(-d $path, '$Config{libpth} is a valid dir');
+}
diff --git a/win32/Makefile b/win32/Makefile
index 62d6ea8..bb99fd8 100644
--- a/win32/Makefile
+++ b/win32/Makefile
@@ -207,7 +207,13 @@ CCTYPE		= MSVC60
 # so you may have to set CCHOME explicitly (spaces in the path name should
 # not be quoted)
 #
+!IF "$(MSVCDIR)" != ""
 CCHOME		= $(MSVCDIR)
+!ELSEIF "$(VCINSTALLDIR)" != ""
+CCHOME		= $(VCINSTALLDIR)
+!ELSE
+!ERROR CCHOME has not been set and could not be automatically determined.
+!ENDIF
 CCINCDIR	= $(CCHOME)\include
 CCLIBDIR	= $(CCHOME)\lib
 
diff --git a/win32/makefile.mk b/win32/makefile.mk
index 7937c5c..c069c6b 100644
--- a/win32/makefile.mk
+++ b/win32/makefile.mk
@@ -228,8 +228,10 @@ CCTYPE		*= GCC
 #
 .IF "$(CCTYPE)" == "GCC"
 CCHOME		*= C:\MinGW
-.ELSE
+.ELIF "$(MSVCDIR)" != ""
 CCHOME		*= $(MSVCDIR)
+.ELIF "$(VCINSTALLDIR)" != ""
+CCHOME		*= $(VCINSTALLDIR)
 .ENDIF
 
 #
@@ -1043,6 +1045,10 @@ CHECKDMAKE :
 	@echo Please get the latest dmake from http://search.cpan.org/dist/dmake/
 	@exit 1
 .ENDIF
+.IF "$(CCHOME)" == ""
+	@echo CCHOME has not been set and could not be automatically determined.
+	@exit 1
+.ENDIF
 
 $(GLOBEXE) : perlglob$(o)
 .IF "$(CCTYPE)" == "GCC"
-- 
1.8.0.msysgit.0


```
</details>

**Comment by p5pRT** [comment]
### From @tonycoz

On Tue Apr 28 17&#8203;:27&#8203;:53 2015\, bulk88 wrote&#8203;:
> Since there have been no comments for this bug\, I have written a patch
> for review\. Mingw is likely to fail
> the added tests so that is why I added WIP\.


> Mingw is likely to fail the tests unless CCHOME is changed manually\, which 
> it didn't need to be changed previously\. 

If you don't set CCHOME t/op/taint\.t will probably fail\, since it needs the files
copied in the test\-prep\-gcc rule\.

Tony

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @steve-m-hay

I think this ticket can be closed in the light of commits 74102a88afc9d8f1973892ce66cf9a83e79d043a and 6f669bf58bbfaf8d7ef367574be9252df84c08d1?

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

I think this ticket can be closed in the light of commits 74102a88afc9d8f1973892ce66cf9a83e79d043a and 6f669bf58bbfaf8d7ef367574be9252df84c08d1?

**Comment by tonycoz** [comment]
> 
> ### From @steve-m-hay
> 
> I think this ticket can be closed in the light of commits [74102a8](https://github.com/Perl/perl5/commit/74102a88afc9d8f1973892ce66cf9a83e79d043a) and [6f669bf](https://github.com/Perl/perl5/commit/6f669bf58bbfaf8d7ef367574be9252df84c08d1)?

Yep, I get reasonable values with MSVC 2019:

```
C:\Users\Tony\dev\perl\git\perl\win32>..\perl -I..\lib -V:libpth
libpth='"C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.23.28105\\lib\x64"';

C:\Users\Tony\dev\perl\git\perl\win32>..\perl -I..\lib -V:incpath
incpath='"C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.23.28105\\include"';
```
closing.
